src/napari_imagej/widgets/menu.py

```diff
@@ -227,8 +228,13 @@ def get_chosen_layer(self) -> None:
                 self._add_image(image)
 
     def get_active_layer(self) -> None:
-        # Choose the active DatasetView
-        view = ij().get("net.imagej.display.ImageDisplayService").getActiveDatasetView()
+        # HACK: Sync ImagePlus before transferring
+        if ij().legacy and ij().legacy.isActive():
+            ij().py.sync_image(ij().WindowManager.getCurrentImage())
+        # Get the active view from the active image display
+        ids = ij().get("net.imagej.display.ImageDisplayService")
+        display = ids.getActiveImageDisplay()
+        view = ids.getActiveDatasetView(display)
```

Comment:
Is there a reason you switched from `ids.getActiveDatasetView()` to the more verbose two-liner here? The `display` variable is not used later, is it?

Reply:
Good question, I'm not sure.

Switched back to `ids.getActiveDatasetView()` in 750939f